lib/iris/_merge.py

```diff
@@ -1228,17 +1227,23 @@ def merge(self, unique=True):
                 data = self._skeletons[group[offset]].data
                 # Ensure the data is represented as a dask array and
                 # slot that array into the stack.
-                if is_lazy_data(data):
+                if iris.util.is_dask_array(data):
                     all_have_data = False
                 else:
-                    data = as_lazy_data(data)
+                    if isinstance(data, np.ma.MaskedArray):
+                        if np.ma.is_masked(data):
+                            if data.dtype.kind == 'i':
+                                data = data.astype('f8')
+                            data[data.mask] = np.nan
+                        data = data.data
```

Comment:
@marqh Okay, I've been waiting to see this kinda pattern appear in the code base, and here it is ...

On `master`, this code block replaces
```python
data = biggus.NumpyArrayAdapter(data)
```
which takes the concrete data and wraps it up as a biggus thing, so that all data is on an equal footing i.e. a biggus array.

For convenience, we could opt to do something similar, when we want a concrete array to be a lazy thing and ensure that masking is deal with appropriately. This could be achieved by having a new `DataProxy` that, akin to the  `PPDataProxy` and `NetCDFDataProxy` would honour the task of massaging the data into `nan` format if necessary.

So, for above we would have something along the lines of
```python
data = da.from_array(DataProxy(data), chunks=data.shape)
```

... and when (if) dask supports masked arrays, this line would become (something like)
```python
data = da.from_array(data, chunks=data.shape)
```

That's pretty attractive to me. It compartmentalizes the data munging to `nan`, and the `DataProxy` pattern can easily be employed else where to consistently do the job of lazifying masked data.